parser/parser.go

```diff
@@ -1096,7 +1103,7 @@ func parseAddressValue(addressText string) (
 	var endOfUri int
 	var startOfParams int
 	if addressText[0] != '<' {
-		if displayName != nil {
+		if _, ok := displayName.(base.String); ok {
```

Comment:
Why the switch to a type assertion rather than a type switch here? We should probably be consistent; but if you have even a vague justification I'm happy to leave it be :)